src/main/java/mate/academy/internetshop/model/Bucket.java

```diff
@@ -1,12 +1,52 @@
 package mate.academy.internetshop.model;
 
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.MapsId;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.Transient;
 import java.util.List;
 
+@Entity
+@Table(name = "buckets")
 public class Bucket {
+
+    @Id
+    @Column(name = "user_id")
     private Long id;
+
+    @ManyToMany(cascade = CascadeType.ALL)
+    @LazyCollection(LazyCollectionOption.FALSE)
+    @JoinTable(name = "buckets_items",
+            joinColumns = @JoinColumn(name = "bucket_id"),
+            inverseJoinColumns = @JoinColumn(name = "item_id"))
     private List<Item> items;
+
+    @Transient
     private Long userId;
 
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "user_id")
+    @MapsId
```

Comment:
do we need this annotation?

Reply:
My unidirectional shared primary key working only this way